### PR TITLE
Moving the `xfail` in the `test_io` file from the test code to the decorator

### LINKED ISSUE #3295: Some marked as xfail tests pass in "cloud" mode
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **Modin version** (`modin.__version__`): 0.10.0
- **Python version**: 3.8.10 64-bit

### Describe the problem
There are tests that are marked as xfail in the `"modin in the cloud"` mode, which actually pass. One needs to find such tests and enable them, for maximum verification of the functionality of the mode.

Some tests are marked as `xfail` in the test code itself. Because of how `pytest.xfail` works (it actually never runs the test), we will not know that the xfail-ed test is really passing (for decorator-implemented `xfail` we receive a `xpass` error). In such tests, you need to transfer xfail from the test itself to the decorator.

### PR DESCRIPTION
**Pull request by Lozovskii-Aleksandr**
## What do these changes do?

- [X] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [X] passes `flake8 modin`
- [X] passes `black --check modin`
- [X] signed commit with `git commit -s` 
- [X] Resolves #3295?
- [X] tests added and passing
- [X] module layout described at `docs/developer/architecture.rst` is up-to-date 

In the file `modin/pandas/test/test_io.py` many tests are marked as `xfail` in `"modin in the cloud"` mode. In addition, these `xfail` are located in the `if` blocks in the test body itself. This is wrong, as if this test somehow works, then there is no way to find out about it.

I looked through the entire file and rearranged `xfail` into the decorators. More information about the solution is written in https://github.com/modin-project/modin/issues/3295.

### CODE REVIEW

File: modin/pandas/test/test_io.py [hunk 1]
```diff
@@ -1117,7 +1081,7 @@ def get_internal_df(df):
 
     @pytest.mark.xfail(
         condition="config.getoption('--simulate-cloud').lower() != 'off'",
-        reason="The reason of tests fail in `cloud` mode is unknown for now - issue #2340",
+        reason="The reason of tests fail in `cloud` mode is unknown for now - issue #3264",
```
Comment (vnlitvinov) [inline modin/pandas/test/test_io.py:1084]: why did you change the reference? let's leave it as it was until we figure out more details

File: modin/pandas/test/test_io.py [hunk 2]
```diff
@@ -1160,7 +1124,7 @@ def wrapped_read_table(file, method):
...
```
Comment (vnlitvinov) [inline modin/pandas/test/test_io.py:1127]: same here and in other places mentioning 3264 I guess...

**Review by vnlitvinov — COMMENTED** [review]
Thanks for the efforts! Left a few comments.

File: modin/pandas/test/test_io.py (hunk 2 above)
Comment (Lozovskii-Aleksandr) [reply modin/pandas/test/test_io.py:1127]: I will return the old reference (https://github.com/modin-project/modin/issues/2340) to where I changed it. Where there was no reference before, I will specify https://github.com/modin-project/modin/issues/3264, since this problem contains all the errors in 'modin in the cloud'. This solution will be more convenient, because when we investigate the error, we will get into a problem where there is already a list of all the added errors.

**Review by vnlitvinov — APPROVED** [review]
LGTM. Given our current flaky CI (most of the blame goes to GH actions I think), please re-run the tests - we merge on all green (except if codecov is red, as it's currently very random).

File: modin/pandas/test/test_io.py [hunk 3]
```diff
@@ -2012,6 +2084,10 @@ def test_read_feather(self):
         finally:
             teardown_test_files([unique_filename])
 
+    @pytest.mark.xfail(
+        condition="config.getoption('--simulate-cloud').lower() != 'off'",
+        reason="The reason of tests fail in `cloud` mode is unknown for now - issue #2340",
```
[the remark this answers is not in the document]
Comment (Lozovskii-Aleksandr) [reply modin/pandas/test/test_io.py:2089]: Tests that are marked https://github.com/modin-project/modin/issues/2340 are already linked to this error. If you assign these tests to https://github.com/modin-project/modin/issues/3264, you may have to search for the error again. I haven't checked to see if this error is reproducible now. 

It seems to me that for now it is possible to leave the problem https://github.com/modin-project/modin/issues/2340, where it is already indicated. When we fix these errors, we will specify the error number exactly (or remove xfail as we will solve the problem).

File: modin/pandas/test/test_io.py [hunk 4]
```diff
@@ -2056,6 +2132,10 @@ def test_read_pickle(self):
         finally:
             teardown_test_files([unique_filename])
 
+    @pytest.mark.xfail(
+        condition="config.getoption('--simulate-cloud').lower() != 'off'",
+        reason="There is no point in reading from local files.",
```
[the remark this answers is not in the document]
Comment (Lozovskii-Aleksandr) [reply modin/pandas/test/test_io.py:2137]: I'm not sure if it's correct to read/write to local files remotely? It may be worth creating a separate problem to skip tests where reading/writing to local files is carried out in the `"modin in the cloud"` mode. And in this problem, correct the reason and replace `xfail` with `skip`.

[the remark this answers is not in the document]
Comment (YarShev) [reply modin/pandas/test/test_io.py:2137]: > I'm not sure if it's correct to read/write to local files remotely?

We read/write to local files for `ExperimentalPandasOnCloudpython` backend.

File: modin/pandas/test/test_io.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Lozovskii-Aleksandr) [reply modin/pandas/test/test_io.py:2089]: Yes, I agree. I thought that https://github.com/modin-project/modin/issues/2340 was not the reason given by me. 

I'll fix it, thank you.

File: modin/pandas/test/test_io.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Lozovskii-Aleksandr) [reply modin/pandas/test/test_io.py:2137]: Good. Then I will correct for what you suggested.

[the remark this answers is not in the document]
Comment (YarShev) [reply modin/pandas/test/test_io.py]: Actually, I have this test passed locally. Why did you disable it?

File: modin/pandas/test/test_io.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (YarShev) [reply modin/pandas/test/test_io.py]: I still see that you are linking #2340 to different tests such as `test_read_parquet_2462`, `test_read_parquet_pandas_index_partitioned`, `test_read_parquet_pandas_index`, `test_read_parquet_partitioned_directory`.

[the remark this answers is not in the document]
Comment (Lozovskii-Aleksandr) [reply modin/pandas/test/test_io.py]: Yes, you're right, I didn't see it.Thanks.

File: modin/pandas/test/test_io.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (YarShev) [reply modin/pandas/test/test_io.py]: Discussed offline, decided to leave as it is because the test fails when running the whole test_io.py.

**Review by YarShev — APPROVED** [review]
@Lozovskii-Aleksandr , LGTM, thanks!
